src/coap_client_zephyr.c

```diff
@@ -761,6 +762,18 @@ static int golioth_connect(struct golioth_client *client)
         port = colon + 1;
     }
 
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+    char ipv6_addr[32];
+    err = synthesize_ipv6_address(host, ipv6_addr);
+    if (err)
+    {
+        LOG_ERR("Failed to synthesize Golioth Server IPv6 address: %d", err);
+        return err;
+    }
+
+    host = ipv6_addr;
+#endif
```

Comment:
If possible use:
```
if (IS_ENABLED(CONFIG_NET_L2_OPENTHREAD) {
    err = synthesize_ipv6_address(host, ipv6_addr);
    if (err)
    {
        LOG_ERR("Failed to synthesize Golioth Server IPv6 address: %d", err);
        return err;
    }

    host = ipv6_addr;
}
```

so that code will always compile and we will detect build errors quickly during any other regular builds.